query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotBasedRequestHandler.java

```diff
@@ -164,10 +173,24 @@ public QueryCost canHandle(QueryRequest request, ExecutionContext executionConte
       return QueryCost.UNSUPPORTED;
     }
 
-    // TODO: Come up with a way to compute the cost based on request and view definition
-    // Higher columns --> Higher cost,
-    // Finer the time granularity --> Higher the cost.
-    return new QueryCost(0.5);
+//    TODO
+//    if (!this.viewDefinitionSupportsGranularity(viewDefinition, request.getGroupByList())) {
+//      return QueryCost.UNSUPPORTED;
+//    }
+
+    double cost;
+
+    long requestStartTime = getRequestStartTime(request.getFilter());
```

Comment:
I think the logic for computing cost can be abstracted in a method and unit tested?

Reply:
sure, will do that

Reply:
Added unit tests.
Regarding abstracting this in a method - actually, this method itself is for computing the query cost.